src/vstest.console/CommandLine/CommandLineOptions.cs

```diff
@@ -295,14 +295,10 @@ public void AddSource(string source)
                 // Get matching files from file pattern parser
                 matchingFiles = FilePatternParser.GetMatchingFiles(source);
             }
-            catch(TestSourceException ex)
+            catch(TestSourceException ex) when (source.StartsWith("-") || source.StartsWith("/"))
             {
-                if(source.StartsWith("-") || source.StartsWith("/"))
-                {
-                    throw new TestSourceException(
-                        string.Format(CultureInfo.CurrentUICulture, CommandLineResources.InvalidArgument, source));
-                }
-                throw ex;
```

Comment:
We won't throw when `(source.StartsWith("-") || source.StartsWith("/"))` is not satisfied anymore. This should be addressed.

Reply:
> We don't throw when (source.StartsWith("-") || source.StartsWith("/")) is not satisfied anymore.

When the condition is not met, the catch block doesn't apply and the exception just throws, right?

Reply:
Yep, you're right :)